### PR TITLE
Client.connect: Handle empty response

### PR DESCRIPTION
**Pull request by aknuds1**
Handle empty response when connecting.

### CODE REVIEW

**Comment by vasco-santos** [comment]
Thanks for the PR @aknuds1 .

Can you please rebase this PR as well?

File: src/index.js [hunk 1]
```diff
@@ -161,10 +161,13 @@ class Client {
     }
 
     const message = await this.send(request).first()
-    const response = Response.decode(message)
+    if ((message || '') === '') {
```
Comment (vasco-santos) [inline src/index.js]: Why not validating if we got a message back?

```suggestion
    if (!message) {
```

File: test/index.spec.js [hunk 2]
```diff
@@ -294,6 +294,26 @@ describe('daemon client', function () {
       }
     })
 
+    it('should error if it receives an empty response', async () => {
```
Comment (vasco-santos) [inline test/index.spec.js]:
```suggestion
    it('should error if it receives an undefined response', async () => {
```

**Comment by aknuds1** [comment]
@vasco-santos I pushed a revision, where I also add a test for the case of an empty response (as opposed to `undefined`) being sent back.

File: test/index.spec.js [hunk 3]
```diff
@@ -319,6 +319,38 @@ describe('daemon client', function () {
...
+        client = new Client(defaultMultiaddr)
+        await client.attach()
+        const identify = await client.identify()
+        client.close()
+
+        client = new Client(addr2)
+        await client.attach()
+
+        stub = sinon.stub(client, 'send').returns({
+          first: () => { return '' }
```
Comment (vasco-santos) [inline test/index.spec.js]: A message being `''`, will go into the initial validation: `if (!message) `.

I believe you want to send an errResponse message with an undefined `msg` instead?

Comment (aknuds1) [reply test/index.spec.js]: @vasco-santos This is actually what I want, to test for the case where the response is `''`. It should hit the validation on line 164, so `Response.decode` isn't called. I think `Response.decode` might actually fail if the input is empty.

Comment (vasco-santos) [reply test/index.spec.js]: You are right, the other test already exists! 😅

**Comment by aknuds1** [comment]
Thanks @vasco-santos!

**Comment by vasco-santos** [comment]
FYI @aknuds1 once I merge [libp2p/js-libp2p-daemon-client#16](https://github.com/libp2p/js-libp2p-daemon-client/pull/16), I will do a new release, so that you can use these fixes

**Comment by aknuds1** [comment]
Thanks @vasco-santos!

**Comment by vasco-santos** [comment]
released `0.2.1`

**Comment by aknuds1** [comment]
> released `0.2.1`

@vasco-santos 🎉
